rick-and-morty/src/components/Characters.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Link } from 'react-router-dom'
+import './Characters.css'
+
+class Characters extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {},
+      isLoading: true,
+      hasError: false
+    }
+  }
+
+  componentDidMount() {
+    fetch('https://rickandmortyapi.com/api/character/')
+      .then(response => {
+        return response.json();
+      })
+      .then(data => {
+        this.setState({
+          data: data.results,
+          isLoading: false
+        });
+      })
+      .catch(error => {
+        this.setState({
+          hasError: true,
+          isLoading: false
+        });
+      });
+  }
+
+  render() {
+    if (this.state.isLoading) {
+      return <div>Loading...</div>;
+    }
+
+    if (this.state.hasError) {
+      return <div>ERROR, please reload and try again</div>;
+    }
+
+    let characterId = this.props.match.params;
+    characterId = characterId[Object.keys(characterId)[0]] - 1;
```

Comment:
Alternatively, you can also get the id via `let characterId = this.props.match.params.id`, then subtract one.